cpp/src/io/comp/nvcomp_adapter.cpp

```diff
@@ -148,11 +148,15 @@ size_t batched_decompress_temp_size(compression_type compression,
                                     size_t max_total_uncomp_size)
 {
   size_t temp_size = 0;
-  auto const nvcomp_status =
+  auto nvcomp_status =
     batched_decompress_get_temp_size_ex(
-      compression, num_chunks, max_uncomp_chunk_size, &temp_size, max_total_uncomp_size)
-      .value_or(batched_decompress_get_temp_size(
-        compression, num_chunks, max_uncomp_chunk_size, &temp_size));
+      compression, num_chunks, max_uncomp_chunk_size, &temp_size, max_total_uncomp_size);
+
+  if (nvcomp_status
+      .value_or(nvcompStatus_t::nvcompErrorInternal) != nvcompStatus_t::nvcompSuccess) {
+    nvcomp_status = batched_decompress_get_temp_size(
+      compression, num_chunks, max_uncomp_chunk_size, &temp_size);
+  }
 
   CUDF_EXPECTS(nvcomp_status == nvcompStatus_t::nvcompSuccess,
```

Comment:
Hmm. I can't recall if there is a conversion from `std::optional<nvcompStatus_t>` to `nvcompStatus_t`, for this `==` to work. I fear we might need to repeat the `value_or(nvcompErrorInternal)` again.

Reply:
I just checked. The `==` is fine.

Reply:
Yeah just rechecked manually and it seems `operator==` is defined for `std::optional<T>` and `T`

Reply:
I wonder if it would be clearer to just do something like:
```
if ((nvcomp_status == std::nullopt) || (nvcomp_status.value() != nvcompStatus_t::nvcompSuccess)) {
```
then you can avoid having to use `nvcompStatus_t::nvcompErrorInternal`.